tests/cmdline/commands/test_process.py

```diff
@@ -490,3 +395,100 @@ def test_multiple_processes(self):
+
+    # Make sure that calling any command on a non-existing process id will not except but print an error
+    # To simulate a process without a corresponding task, we simply create a node and store it. This node will not
+    # have an associated task at RabbitMQ, but it will be a valid `ProcessNode` so it will pass the initial
+    # filtering of the `verdi process` commands
+    orphaned_node = WorkFunctionNode().store()
+    non_existing_process_id = str(orphaned_node.pk)
+    for command in [process_pause, process_play, process_kill]:
+        result = cli_runner.invoke(command, [non_existing_process_id])
+        import traceback
+        assert result.exception is None, ''.join(traceback.format_exception(*result.exc_info))
+        assert 'Error:' in result.output
+
+    assert not calc.paused
+    result = cli_runner.invoke(process_pause, [str(calc.pk)])
+    assert result.exception is None, result.output
+
+    # We need to make sure that the process is picked up by the daemon and put in the Waiting state before we start
+    # running the CLI commands, so we add a broadcast subscriber for the state change, which when hit will set the
+    # future to True. This will be our signal that we can start testing
+    waiting_future = Future()
+    filters = kiwipy.BroadcastFilter(
+        lambda *args, **kwargs: waiting_future.set_result(True), sender=calc.pk, subject='state_changed.*.waiting'
+    )
+    runner.communicator.add_broadcast_subscriber(filters)
+
+    # The process may already have been picked up by the daemon and put in the waiting state, before the subscriber
+    # got the chance to attach itself, making it have missed the broadcast. That's why check if the state is already
+    # waiting, and if not, we run the loop of the runner to start waiting for the broadcast message. To make sure
+    # that we have the latest state of the node as it is in the database, we force refresh it by reloading it.
+    calc = load_node(calc.pk)
+    if calc.process_state != plumpy.ProcessState.WAITING:
+        runner.loop.run_until_complete(asyncio.wait_for(waiting_future, timeout=5.0))
+
+    # Here we now that the process is with the daemon runner and in the waiting state so we can starting running
+    # the `verdi process` commands that we want to test
+    result = cli_runner.invoke(process_pause, ['--wait', str(calc.pk)])
+    assert result.exception is None, result.output
+    assert calc.paused
+
+    if cmd_try_all:
+        cmd_option = '--all'
+    else:
+        cmd_option = str(calc.pk)
+
+    result = cli_runner.invoke(process_play, ['--wait', cmd_option])
+    assert result.exception is None, result.output
+    assert not calc.paused
+
+    result = cli_runner.invoke(process_kill, ['--wait', str(calc.pk)])
+    assert result.exception is None, result.output
+    assert calc.is_terminated
+    assert calc.is_killed
+
+    os.kill(daemon.pid, signal.SIGTERM)
```

Comment:
This cleanup will now sometimes not be called if during the test an error occurs. We should create a fixture, let's say `with_daemon` that looks a bit like:
```python
@pytest.fixture
def with_daemon():
    # start the daemon process
    yield
    os.kill(daemon.pid, signal.SIGTERM)
```
This way the `kill` will be guaranteed to be called